csharp/src/Ice/InputStream.cs

```diff
@@ -405,6 +408,70 @@ public uint ReadVarUInt()
             return dict;
         }
 
+        /// <summary>Reads a dictionary from the stream.</summary>
+        /// <param name="minKeySize">The minimum size of each key of the dictionary, in bytes.</param>
+        /// <param name="withBitSequence">When true, null dictionary values are encoded using a bit sequence.</param>
+        /// <param name="keyReader">The input stream reader used to read each key of the dictionary.</param>
+        /// <param name="valueReader">The input stream reader used to read each non-null value of the dictionary.
+        /// </param>
+        /// <returns>The dictionary read from the stream.</returns>
+        public Dictionary<TKey, TValue?> ReadDictionary<TKey, TValue>(
+            int minKeySize,
+            bool withBitSequence,
+            InputStreamReader<TKey> keyReader,
+            InputStreamReader<TValue> valueReader)
+            where TKey : notnull
+            where TValue : class
+        {
+            int sz = ReadAndCheckSeqSize(minKeySize);
```

Comment:
Shouldn't this check for the value size if withBitSequence = false? I suppose we don't really use it with false?

Reply:
Yes, in theory we should include the minValueSize when withBitSequence is false, but in practice this method is never called with withBitSequence=false. If you want to read a dictionary without a bit sequence, use another `ReadDictionary` overload without this withBitSequence parameter.